From 0d052d64480a30e83fcdda80f4774624e044beb7 Mon Sep 17 00:00:00 2001
From: Felicia Lim <flim@google.com>
Date: Thu, 28 Jul 2016 15:21:19 +0200
Subject: Ensure that NLSF cannot be negative when computing a min distance
 between them

b/31607432

Signed-off-by: Jean-Marc Valin <jmvalin@jmvalin.ca>
(cherry picked from commit d9d5ac4027c5ee1da2ff1c6572ae3b35a845db19)
---
 silk/NLSF_stabilize.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/silk/NLSF_stabilize.c b/silk/NLSF_stabilize.c
index 1fa1ea3..8f3426b 100644
--- a/silk/NLSF_stabilize.c
+++ b/silk/NLSF_stabilize.c
@@ -130,7 +130,7 @@ void silk_NLSF_stabilize(
 
         /* Keep delta_min distance between the NLSFs */
         for( i = 1; i < L; i++ )
-            NLSF_Q15[i] = silk_max_int( NLSF_Q15[i], NLSF_Q15[i-1] + NDeltaMin_Q15[i] );
+            NLSF_Q15[i] = silk_max_int( NLSF_Q15[i], silk_ADD_SAT16( NLSF_Q15[i-1], NDeltaMin_Q15[i] ) );
 
         /* Last NLSF should be no higher than 1 - NDeltaMin[L] */
         NLSF_Q15[L-1] = silk_min_int( NLSF_Q15[L-1], (1<<15) - NDeltaMin_Q15[L] );
-- 
cgit v0.12

